### PR TITLE
Correctly record opsbehind prop in events

### PR DESCRIPTION
**Pull request by jatgarg**
## Description

Item: https://dev.azure.com/fluidframework/internal/_workitems/edit/4662

We found that Data_opsbehind is not correct when the connection is established in parallel to snapshot fetch as then the lastProcessedSeq number would just be 0 and opsbehind would just be lastKnownSeqNumber - 0 instead of lastKnownSeqNumber - snapshotseqnumber.

Also add opsbehind prop in connected event in case of first connection to see what was the diff between lastProcesssedseqnumber and snapshotseqnumber.

### CODE REVIEW

**Review by NicholasCouri — COMMENTED** [review]
>'					opsBehind = checkpointSequenceNumber - this.deltaManager.lastSequenceNumber;  '

In the NoJoinOp query we use deltaManager.initialSequenceNumber instead of lastSequenceNumber. Should we update our query ?

File: packages/loader/container-loader/src/container.ts [hunk 1]
```diff
@@ -1993,10 +1995,19 @@ export class Container
 				durationFromDisconnected =
 					time - this.connectionTransitionTimes[ConnectionState.Disconnected];
 				durationFromDisconnected = TelemetryLogger.formatTick(durationFromDisconnected);
+				// On first connection, also note how much gap was there between last processed seq number which would
+				// mostly be JoinOp and the snapshot seq number.
+				if (this._snapshotSeqNumber !== undefined && this.firstConnection) {
+					opsBehind = this.deltaManager.lastSequenceNumber - this._snapshotSeqNumber;
```
Comment (vladsud) [inline packages/loader/container-loader/src/container.ts:2001]: Not sure why you are using this.deltaManager.lastSequenceNumber over this.deltaManager.lastKnownSeqNumber. I think later is correct in all cases

Comment (vladsud) [reply packages/loader/container-loader/src/container.ts:2001]: BTW, I'd use the same formular (checkpointSequenceNumber - this.deltaManager.lastSequenceNumber) in all cases - it looks correct to me in a sense of intention - we are recording how far we are behind at the moment of connection. The only thing that is wrong in main - lastSequenceNumber = 0 when connection established before we loaded snapshot.

I'd actually do something like that:

```ts
if (this.deltaManager.hasCheckpointSequenceNumber && this._lifecycleState == "loaded") {
    opsBehind = checkpointSequenceNumber - this.deltaManager.lastSequenceNumber;
}
```

I'd probably not record anything on connected event. We are never behind for "write" connections. For "read" connections we might find it useful to record where we are, but same will be captured for CatchingUp event, and in future we plan to switch "read" to behave like "write", so not much value for this code to record anything.

**Review by NicholasCouri — APPROVED** [review]

**Comment by jatgarg** [comment]
> > '					opsBehind = checkpointSequenceNumber - this.deltaManager.lastSequenceNumber;  '
> 
> In the NoJoinOp query we use deltaManager.initialSequenceNumber instead of lastSequenceNumber. Should we update our query ?

I mean it has the same issue. If the snapshot is loaded in parallel to making a connection, then it will be 0, otherwise it will be same as lastSeqNumber. We may want to eliminate cases entirely in which connection was made in parallel.

**Pull request merged**